### PR TITLE
Upgrade for Xcode 12

### PR DESCRIPTION
**Pull request by jkmassel**
This PR updates the `pod` for Xcode 12 and Apple Silicon and modernizes the CI by:
- Using the latest version of CocoaPods for XC12/AS compatibility
- Using the latest version of bundler to avoid recently security concerns
- Using `fastlane` for CI actions instead of the CircleCI-specific DSL
- Updating the `.ruby-version` to a newer release to align with the version available in CircleCI

**To Test:**
- The automated tests will pass (one can't get status reported because it's removed in favour of a more descriptive name)
- Branch from this PR and create a new tag (that doesn't already exist). Ensure that the publish succeeds, and note the new build message in `#build-and-ship`.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,33 +2,84 @@ version: 2.1
...
+      - fix-image
+      - checkout
+      - ios/install-dependencies:
+          bundle-install: true
+          cache-prefix: cache-prefix-{{ checksum "wpxmlrpc.podspec" }}-publish-podspec-v1 # Effectively disables caching when publishing
+      - run:
+          name: Publish Podspec
+          command: |
+            ruby --version
+            bundle exec fastlane publish_podspec
```
Comment (AliSoftware) [inline .circleci/config.yml:77]: Why not use the commands from the ios orb directly (`ios/publish-podspec` et al.)?

File: fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    slack(
+      message: ":tada: #{podspec.module_name} #{podspec.version} successfully published to CocoaPods trunk!\nIt will take a few minutes for this version to be deployed to the CocoaPods CDN.",
+      success: true,
+      slack_url: ENV['PODS_SLACK_WEBHOOK'],
+      payload: [
+          "Build URL" => ENV["CIRCLE_BUILD_URL"],
+       ]
+    )
+  end
```
Comment (AliSoftware) [inline fastlane/Fastfile:39]: The orb already has all the logic for this, so I wonder what's the benefit of replicating this as a lane? 🤔

File: .circleci/config.yml [hunk 3]
```diff
@@ -2,33 +2,84 @@ version: 2.1
 
 orbs:
   # Using 1.0 of our Orbs means it will use the latest 1.0.x version from https://github.com/wordpress-mobile/circleci-orbs
-  ios: wordpress-mobile/ios@1.0 
+  ios: wordpress-mobile/ios@1.0
+
+# YAML anchors for some common/repeated values
+x-common-params:
+  - &xcode-version "12.2.0"
```
Comment (mokagio) [inline .circleci/config.yml]: Would Xcode 12.4 ( work as well? Or is there some compatibility issue with the rest of the tooling?

**Review by mokagio — APPROVED** [review]
✅ 

For the record, after this PR was opened, we had an internal discussion and agreed that, while we've been relying on the CircleCI orb to DRY our config files so far, we'd rather make our automation CI-independent, which motivates the introduction of new logic in here even thought we could have done that with the orb commands.

File: .circleci/config.yml (hunk 3 above)
Comment (jkmassel) [reply .circleci/config.yml]: Yeah, good call. Seems to work great, so let's roll with it!

**Pull request merged**